include/Utilities/MathUtilities.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+#include "../Geometrics/Math/EquationInterface.h"
+#include "../Geometrics/Math/QuadraticEquation.h"
+#include "../pch.h"
+
+namespace Barta::Utils {
+std::unique_ptr<EquationInterface> createQuadraticEquation(
+    float A,
+    float B,
+    float C
+) {
+    return std::unique_ptr<EquationInterface>(new QuadraticEquation(A, B, C));
```

Comment:
There is no use of heap allocation here. Note that this code executes each main loop iteration so we want it to be as fast as possible. It is enough to return just `QuadraticEquation(...)` and the copy elision will take care of the rest ;)